Balance tweaks, QOL & some cleanup

StrawWagen opened the pull request:
All panels now close when you click off of them, press use again, or open the Q menu.

Turrets have been split into two categories, heavy turrets and light turrets
Based off testing, gives a use to the crap turrets, while limiting the super strong ones.

Light turrets (smg, pistol, plinker)
spawn with ammo & a battery already attached. 
Max 4 light turrets

Heavy turrets( tesla, grenade, machinegun, shotgun, sniper )
They're how turrets were before, you have to manually attach batteries/ammo 
They now drain battery a lot faster ( hook them up to a generator! ) 
Max 2 heavy turrets.

NPCS now target/fight the turrets.

Grenade turret now has a proper tracer, can actually figure out where you're being shot from.

Made generator turning on/off less vague

Spam the "broken" effect less ( affects implanter, teslaturret, normal turret )

Paintcan now defaults to last applied color.

Can now swap out batteries when they're below 25% charge ( had to be completely empty before ) 
Same with ammo. ( has to be below five rounds )

Grenade turret damage, firerate buff.
Range, HP, max ammo, grenade flight speed nerf.

Sniper turret range, firerate, damage buff.
Small HP nerf

Barbed wire no longer takes ACF damage.

StrawWagen committed:
Balance tweaks, QOL & some cleanup

All panels now close when you click off of them, press use again, or open the Q menu.

Turrets have been split into two categories, heavy turrets and light turrets
Light turrets (smg, pistol, plinker) spawn with ammo & a battery already attached.
Max 4 light turrets
Heavy turrets ( tesla, grenade, machinegun, shotgun, sniper )
are how turrets were before, you have to manually attach batteries/ammo
Heavy turrets now drain battery a lot faster ( hook them up to a generator! )
Max 2 heavy turrets.

NPCS now target/fight the turrets.

Grenade turret now has a proper tracer, can actually figure out where you're being shot from.

Made generator turning on/off less vague

Spam the "broken" effect less ( affects implanter, teslaturret, normal turret )

Paintcan now defaults to last applied color.

Can now swap out batteries when they're below 25% charge ( had to be completely empty before )
Same with ammo. ( has to be below five rounds )

Grenade turret damage, firerate buff.
Range, HP, max ammo, grenade flight speed nerf.

Sniper turret range, firerate, damage buff.
Small HP nerf

StrawWagen committed:
Change spawnicons to png

StrawWagen committed:
Give ball buster kill credit to whoever arms it

wrefgtzweve commented on materials/entities/ent_jack_aidcomm.png:
This file can be removed alongside other pngs that dont exist for our current entities

StrawWagen replied:
Done, removed ents too

wrefgtzweve commented on lua/autorun/client/jid_client.lua:
```diff
@@ -16,3 +16,43 @@ killicon.Add( "ent_jack_seamine", "vgui/hud/jid_seamine_killicon", iconColor )
 killicon.Add( "ent_jack_boundingmine", "vgui/hud/jid_boundingmine_killicon", iconColor )
 killicon.Add( "ent_jack_teslasentry", "vgui/hud/jid_teslasentry_killicon", iconColor )
+
+local input_IsKeyDown = input.IsKeyDown
```
Localizing these seems rather pointless

wrefgtzweve commented on lua/effects/eff_jack_rocketthrust/init.lua:
```diff
@@ -1,3 +1,6 @@
+local math = math
```
I dont see the point of localizing `math` here, same for `VectorRand` and in other files.

wrefgtzweve commented on lua/entities/ent_jack_40mmgrenade/init.lua:
```diff
@@ -72,3 +110,16 @@ function ENT:Detonate()
+    local smokeTrail = self.SmokeTrail
+    if IsValid( smokeTrail ) then
+        SafeRemoveEntityDelayed( smokeTrail, smokeTrailLifetime )
+
```
Empty newline should be removed and added after the `end` instead, same for the tracerTrail

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_barbedwire/init.lua:
```diff
@@ -190,4 +190,10 @@ function ENT:Think()
+    -- barbed wire can't be broken by ACF
+    return false
+
 end
```
Add a empty newline to the end of this file

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_seamine/init.lua:
```diff
@@ -125,3 +135,9 @@ function ENT:Use( activator )
+function ENT:ACF_PreDamage()
+    -- ball busters are immune
+    return false
+
```
Newline should be removed

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_turret_base/init.lua:
```diff
@@ -133,6 +136,8 @@ function ENT:Initialize()
     self:SetMoveType( MOVETYPE_VPHYSICS )
     self:SetSolid( SOLID_VPHYSICS )
     self:SetUseType( SIMPLE_USE )
+    -- allow npcs to target me
```
i prefer if you left comments like these after the call
```lua
self:AddFlags( FL_OBJECT ) -- allow npcs to target the turret
```

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_turret_base/init.lua:
```diff
@@ -157,6 +162,24 @@ function ENT:Initialize()
+
+    timer.Simple( 0, function()
+        if not IsValid( self ) then return end
+        -- used by "heavy" turrets
```
Comment should be on the same line as the code it affects

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_turret_base/init.lua:
```diff
@@ -797,31 +832,17 @@ function ENT:FireShot()
-    if self.AmmoType == "7.62x51mm" or self.AmmoType == ".338 Lapua Magnum" then
-        sound.Play( self.NearShotNoise, ShootPos + Vector( 0, 0, 1 ), 75, self.ShootSoundPitch + 10 )
+    local filterAllPlayers = RecipientFilter()
+    filterAllPlayers:AddAllPlayers()
```
Why do this when EmitSound by default respects PAS? networking it to all players makes no logical sense to me. If sounds didn't properly work before use :AddPVS

StrawWagen replied:
PAS breaks alot with high soundlevel, seems to just not get to players on bigcity sometimes, etc.
Added this so players can always hear where they're getting shot from, no confusion
Changed to PVS 👍

wrefgtzweve commented on lua/entities/ent_jack_turret_base/init.lua:
```diff
@@ -797,31 +832,17 @@ function ENT:FireShot()
+    self:EmitSound( self.NearShotNoise, 70 + rangeLevelOffset, self.ShootSoundPitch, 1, CHAN_WEAPON )
+    -- semi-global sound so players know where they're getting shot from
+    self:EmitSound( self.FarShotNoise, 90 + rangeLevelOffset, self.ShootSoundPitch - 10, 1, CHAN_WEAPON, 0, 0, filterAllPlayers )
+    -- play extra sounds to make it feel punchier.
```
Comments should be on the same line as affected code

StrawWagen replied:
👍

wrefgtzweve commented on lua/entities/ent_jack_turret_grenade/init.lua:
```diff
@@ -136,8 +135,13 @@ function ENT:FireShot()
-        TargPos = TargPos + Vector( 0, 0, distPow / 60 )
+        if Dist <= 1000 and CurrTarg:IsPlayer() or CurrTarg:IsNPC() then
+            TargPos = CurrTarg:GetPos() -- aim at feet if they're close
+
```
Newline can be removed

StrawWagen replied:
👍

wrefgtzweve reviewed (commented):
Mostly style but here and there ive had some questions about the code.

StrawWagen committed:
Apply suggestions from code review.

Removed unused ents, materials. models, sounds
Tweaked style all over
Use AddPVS instead of AddAllPlayers for turret far shot sounds

StrawWagen committed:
Increase ball buster comedy

Also buffed damage

wrefgtzweve commented:
I don't know if you should remove the entity files for the unused ents, if you think we'll never use them feel free to. Keeping them for future use however is no a issue, my initial remove comment was just about not having materials/content/downloadsize for unused stuff.

StrawWagen commented:
Only extra ents left were the missile turrets, would've been retreading old ground gameplay wise, removin em is no issue

wrefgtzweve commented:
👍
